libp2p/switch.nim

```diff
@@ -351,6 +364,7 @@ proc newSwitch*(peerInfo: PeerInfo,
   result.identity = identity
   result.muxers = muxers
   result.secureManagers = initTable[string, Secure]()
+  result.dialedPubSubPeers = initHashSet[string]()
```

Comment:
this would be a good opportunity to clean up the init here with `Switch(a: b, c: d)`

Reply:
I was gonna clean that up in one swipe in a separate effort, I would merge this as is since it's affecting integration with `nbc`.